src/net.cpp

```diff
@@ -904,7 +903,20 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
         }
     }
 
-    bool whitelisted = hListenSocket.whitelisted || IsWhitelistedRange(addr);
+    NetPermissionFlags permissionFlags = NetPermissionFlags::PF_NONE;
+    hListenSocket.AddSocketPermissionFlags(permissionFlags);
+    AddWhitelistPermissionFlags(permissionFlags, addr);
+    const bool noban = NetPermissions::HasFlag(permissionFlags, NetPermissionFlags::PF_NOBAN);
+    bool legacyWhitelisted = false;
+    if (NetPermissions::HasFlag(permissionFlags, NetPermissionFlags::PF_ISIMPLICIT)) {
+        NetPermissions::ClearFlag(permissionFlags, PF_ISIMPLICIT);
+        if (gArgs.GetBoolArg("-whitelistforcerelay", false)) NetPermissions::AddFlag(permissionFlags, PF_FORCERELAY);
+        if (gArgs.GetBoolArg("-whitelistrelay", false)) NetPermissions::AddFlag(permissionFlags, PF_RELAY);
```

Comment:
Why are the fallbacks set to `false` and not the `DEFAULT_` value?

Reply:
Because the constant is in a different file which is not in imported into the scope of this file.

Looking at other `GetBoolArg` references in this file, you can notice that all of them have the same issue.

Reply:
So is it safe to do that? Under what circumstances would it lead to bugs?

Reply:
I guess you are fixing it in #16631?

Reply:
@MarcoFalke yes, I misunderstood your question. I thought you were asking me to use the `DEFAULT_` const instead of hard coding the value. But you were saying that the `DEFAULT_` is not the same as what I hardcoded.